static/app/components/profiling/flamegraph/flamegraphPreview.tsx

```diff
@@ -0,0 +1,244 @@
+    if (!flamegraphCanvas) {
+      return null;
+    }
+
+    const canvasView = new CanvasView({
+      canvas: flamegraphCanvas,
+      model: flamegraph,
+      options: {barHeight: flamegraphTheme.SIZES.BAR_HEIGHT},
+      mode: 'anchorBottom',
+    });
+
+    canvasView.setConfigView(
+      computePreviewConfigView(
+        flamegraph,
+        canvasView.configView,
+        relativeStartTimestamp * 1e9,
+        relativeStopTimestamp * 1e9
+      )
+    );
+
+    return canvasView;
+  }, [
+    flamegraph,
+    flamegraphCanvas,
+    flamegraphTheme,
+    relativeStartTimestamp,
+    relativeStopTimestamp,
+  ]);
+
+  const flamegraphCanvases = useMemo(() => [flamegraphCanvasRef], [flamegraphCanvasRef]);
+
+  useResizeCanvasObserver(
+    flamegraphCanvases,
+    canvasPoolManager,
+    flamegraphCanvas,
+    flamegraphView
+  );
+
+  const flamegraphRenderer = useMemo(() => {
+    if (!flamegraphCanvasRef) {
+      return null;
+    }
+
+    return new FlamegraphRenderer2D(flamegraphCanvasRef, flamegraph, flamegraphTheme, {
+      draw_border: true,
+    });
+  }, [flamegraph, flamegraphCanvasRef, flamegraphTheme]);
+
+  const textRenderer = useMemo(() => {
+    if (!flamegraphCanvasRef) {
+      return null;
+    }
+    return new FlamegraphTextRenderer(flamegraphCanvasRef, flamegraphTheme, flamegraph);
+  }, [flamegraph, flamegraphCanvasRef, flamegraphTheme]);
+
+  useEffect(() => {
+    if (!flamegraphCanvas || !flamegraphView || !flamegraphRenderer || !textRenderer) {
+      return undefined;
+    }
+
+    const clearOverlayCanvas = () => {
+      textRenderer.context.clearRect(
+        0,
+        0,
+        textRenderer.canvas.width,
+        textRenderer.canvas.height
+      );
+    };
+
+    const drawRectangles = () => {
+      flamegraphRenderer.draw(
+        flamegraphView.fromTransformedConfigView(flamegraphCanvas.physicalSpace)
+      );
+    };
+
+    const drawText = () => {
+      textRenderer.draw(
+        flamegraphView.toOriginConfigView(flamegraphView.configView),
+        flamegraphView.fromTransformedConfigView(flamegraphCanvas.physicalSpace)
+      );
+    };
+
+    scheduler.registerBeforeFrameCallback(clearOverlayCanvas);
+    scheduler.registerBeforeFrameCallback(drawRectangles);
+    scheduler.registerBeforeFrameCallback(drawText);
+
+    scheduler.draw();
+
+    return () => {
+      scheduler.unregisterBeforeFrameCallback(clearOverlayCanvas);
+      scheduler.unregisterBeforeFrameCallback(drawRectangles);
+      scheduler.unregisterBeforeFrameCallback(drawText);
+    };
+  }, [flamegraphRenderer, flamegraphView, flamegraphCanvas, scheduler, textRenderer]);
+
+  return (
+    <CanvasContainer>
+      <Canvas ref={setFlamegraphCanvasRef} />
+    </CanvasContainer>
+  );
+}
+
+export function computePreviewConfigView(
+  flamegraph: FlamegraphModel,
+  configView: Rect,
+  relativeStartNs: number,
+  relativeStopNs: number
+): Rect {
+  const frames: FlamegraphFrame[] = flamegraph.root.children.slice();
```

Comment:
There is an early return opportunity here because flamegraphs track the max depth already. If flamegraph depth < view height, you can just return the computed max